libp2p/protocols/kademlia/keys.nim

```diff
@@ -0,0 +1,48 @@
+import ../../peerid
+import ./consts
+import chronicles
+import stew/byteutils
+
+type
+  KeyType* {.pure.} = enum
+    Undefined
```

Comment:
why have `Undefined` as possibility? can it be avoided?

Reply:
I guess more than Undefined, it should be called NotHashed, or similar.